user_tools/src/spark_rapids_pytools/cloud_api/databricks_azure.py

```diff
@@ -174,38 +174,28 @@ def _build_cmd_scp_from_node(self, node: ClusterNode, src: str, dest: str) -> st
                        dest]
         return Utils.gen_joined_str(' ', prefix_args)
 
-    def process_instances_description(self, raw_instances_description: str) -> dict:
-        processed_instances_description = {}
-        instances_description = JSONPropertiesContainer(prop_arg=raw_instances_description, file_load=False)
-        for instance in instances_description.props:
-            instance_dict = {}
-            v_cpus = 0
-            memory_gb = 0
-            gpus = 0
+    def _process_instance_description(self, instance_descriptions: str) -> dict:
+        processed_instance_descriptions = {}
+        raw_instance_descriptions = JSONPropertiesContainer(prop_arg=instance_descriptions, file_load=False)
+        for instance in raw_instance_descriptions.props:
             if not instance['capabilities']:
                 continue
-            for item in instance['capabilities']:
-                if item['name'] == 'vCPUs':
-                    v_cpus = int(item['value'])
-                elif item['name'] == 'MemoryGB':
-                    memory_gb = int(float(item['value']) * 1024)
-                elif item['name'] == 'GPUs':
-                    gpus = int(item['value'])
-            instance_dict['VCpuInfo'] = {'DefaultVCpus': v_cpus}
-            instance_dict['MemoryInfo'] = {'SizeInMiB': memory_gb}
-            if gpus > 0:
-                gpu_list = [{'Name': '', 'Manufacturer': '', 'Count': gpus, 'MemoryInfo': {'SizeInMiB': 0}}]
-                instance_dict['GpuInfo'] = {'GPUs': gpu_list}
-            processed_instances_description[instance['name']] = instance_dict
-        return processed_instances_description
-
-    def generate_instances_description(self, fpath: str):
-        cmd_params = ['az vm list-skus',
-                      '--location', f'{self.get_region()}']
-        raw_instances_description = self.run_sys_cmd(cmd_params)
-        json_instances_description = self.process_instances_description(raw_instances_description)
-        with open(fpath, 'w', encoding='UTF-8') as output_file:
-            json.dump(json_instances_description, output_file, indent=2)
+            instance_content = {}
+            gpu_count = 0
+            for elem in instance['capabilities']:
+                if elem['name'] == 'vCPUs':
+                    instance_content['DefaultVCpus'] = int(elem['value'])
+                elif elem['name'] == 'MemoryGB':
+                    instance_content['MemoryInMB'] = int(float(elem['value']) * 1024)
+                elif elem['name'] == 'GPUs':
+                    gpu_count = int(elem['value'])
+            if gpu_count > 0:
+                instance_content['GpuInfo'] = [{'Count': gpu_count}]
```

Comment:
Storing only the GPU count might not be sufficient. We should also parse and store the GPU name.

Reply:
Thanks @parthosa! Databricks azure cli does not provide gpu name in the results. I believe this is okay for now since Q tool is not pulling gpu info from input clusters.

Udated the json key to `VCpuCount`.